### ISSUE TITLE
"outside repository" fatal error

### ISSUE THREAD

**Issue by tiagovignatti** [SRombauts/UE4GitPlugin#34]
Hi! I'm having a problem when I commit into my project, complaining about the files that are used but are external to the repository and live in the engine instead. This is what the Message Log shows when I try to commit:
![gitplugin-error](https://cloud.githubusercontent.com/assets/1163553/20007686/9bdea6a2-a284-11e6-9ead-62c6e4481ac4.png)

I understand what's causing this is the fact that only the project files were set to be part of the git repository and not the ones inside the engine.  But what's the idea to solve this? I thought that the plugin would take care of that. Thank you.

**Comment by SRombauts** [comment]
Hi!

Sorry about that, this is more or less a known problem, depending on how you use the Editor, but I shall try to get rid of this when I get some time to spend into it...

Cheers!

**Comment by nkpfstr** [comment]
I'm encountering this same issue in a 4.16 project. 😢 I started the project and Git repo with 4.16 Preview 1. It worked fine through all the preview releases, but today using the 4.16 stable release I am getting errors for almost every action.

If I launch the editor and click "Submit to Source Control":
```
git-lfs filter-process: git-lfs: command not found
fatal: The remote end hung up unexpectedly
fatal: C:/Program Files/Epic Games/UE_4.16/Engine/Content/: 'C:/Program Files/Epic Games/UE_4.16/Engine/Content/' is outside repository
```

If I turn off source control and attempt to connect again:
```
git-lfs filter-process: git-lfs: command not found
fatal: The remote end hung up unexpectedly
Failed to enable Git source control. You need to initialize the project as a Git repository first.
```

Just thought I'd give you more evidence to work with in addressing this issue. Good luck fixing it! Thanks for all the hard work! 😄

**Comment by SRombauts** [comment]
Okay, trying to reproduce this with UE4.16, git version 2.13.0.windows.1, which include git-lfs/2.1.0 (GitHub; windows amd64; go 1.8.1; git bd2c9987).

Creating a new First Person Blueprint project, using the automatic repository creation and automatic initial commit.

Here is the message I get when I commit a modified asset using right-clic context menu (no problem):

```
[2017.05.25-05.46.53:155][212]LogSourceControl: FGitCheckInWorker: commit successful
[2017.05.25-05.46.53:257][212]SourceControl: Info [master f7c0cf1] test mono checkin
[2017.05.25-05.46.53:257][212]SourceControl: Info  1 file changed, 0 insertions(+), 0 deletions(-)
[2017.05.25-05.46.53:265][212]SourceControl: Info Commited [master f7c0cf1] test mono checkin.
```

Using the global "Submit to Source Control", I get an info message with the error (but it's filtered out as not a real error, hence the "Info" prefix:

```
[2017.05.25-05.49.07:489][185]SourceControl: Info fatal: C:/Program Files/Epic Games/UE_4.16/Engine/Content/: 'C:/Program Files/Epic Games/UE_4.16/Engine/Content/' is outside repository
[2017.05.25-05.49.19:219][185]LogSourceControl: FGitCheckInWorker: commit successful
[2017.05.25-05.49.19:322][185]SourceControl: Info [master 0cd69ac] global submit to source control
[2017.05.25-05.49.19:322][185]SourceControl: Info  1 file changed, 0 insertions(+), 0 deletions(-)
[2017.05.25-05.49.19:333][185]SourceControl: Info Commited [master 0cd69ac] global submit to source control.
```

**Comment by SRombauts** [comment]
@nkpfstr Can you give me some more info, what Git version? How did you create your repository: using the connect screen of the plugin? Or with a GUI tool? Do you happen to have a .gitattributes file? Did you try to setup git LFS at some point?

**Comment by SRombauts** [comment]
Using git version 2.10.0.windows.1 without Git LFS on the same project is working the same.

**Comment by nkpfstr** [comment]
> What Git version?

2.11.0.windows.3

> How did you create your repository?

With the connect screen of the UE4 plugin.

> Do you have a .gitattributes file?

Yes. It's configured for Git LFS to recognize Unreal assets:
```
*.uasset filter=lfs diff=lfs merge=lfs -text
*.umap filter=lfs diff=lfs merge=lfs -text
```

> Did you try to setup Git LFS?

Yes, see above. Worth noting that I setup LFS with GitHub Desktop, which I also use for all my Git operations outside UE4.

> What is the path to your project?

`D:/UnrealProjects/<Project>`

It's on a separate drive; UE4 and Git are both installed on C drive.

**Comment by SRombauts** [comment]
Thanks for the detailed information, I shall try to see if projects on a secondary drive is a problem

**Comment by SRombauts** [comment]
Hi @nkpfstr, so I reproduced your problem by creating a new project with Git+LFS, then uninstalling Git and LFS: the plugin then find another "git.exe" (for me inside SmartGit, for you the one shipped with Github) but for this git.exe there is no associated LFS extension, so any actions now fails!

Can you please confirm that your Git.exe path is not the standard one ("C:\Program Files\Git\bin\git.exe")?

**Comment by nkpfstr** [comment]
That's correct. The Git.exe detected by the UE4 Git plugin is in `C:\Users\Nick\AppData\Local\GitHub\PortableGit_<long-string-of-random-characters>\cmd\git.exe`.

**Comment by SRombauts** [comment]
Ok, so you should be able to override this with the correct (above) path and get things working again :)

Also, there should have been no changes between 4.16 preview and final release, so I am not sure about why you have this problem at this time...

**Comment by SRombauts** [comment]
Just for future reference, this "fatal:" error is only when you have the "Engine Content" displayed in the Content Browser!

LogSourceControl: Warning: RunCommand(status) ReturnCode=128:
fatal: C:/Program Files/Epic Games/UE_4.17/Engine/Content: 'C:/Program Files/Epic Games/UE_4.17/Engine/Content' is outside repository

**Comment by FurryFur** [comment]
Not sure if this is related but I can't get this plugin to work at all with 4.19.2. This is the only error in the message log. It happens when making a new third-person template project and trying to check out any file. The file always fails to checkout, however, the file still gets locked and this error is displayed in the message log. I am trying to use git lfs and file locking with a private github repository.

**Comment by ghildebrandt** [comment]
even though we do not display the engine content in the content browser we get this error on every file after the first push

**Comment by svenfraeys** [comment]
hello,
I am experiencing the same problems at some point.
Here is part of my log, hope this helps.

[2019.02.19-16.15.19:709][298]LogSourceControl: LockedFile(F:/Unreal Projects/Project_Production/Content/mats/NewMaterial2.uasset, Sven)
[2019.02.19-16.15.19:709][298]LogSourceControl: RunCommand: 'git status --porcelain --ignored "F:/4.20.3-customEngine/Engine/Content"'
[2019.02.19-16.15.19:770][302]LogSourceControl: RunCommand(status):

[2019.02.19-16.15.19:770][302]LogSourceControl: Warning: RunCommand(status) ReturnCode=128:
fatal: F:/4.20.3-customEngine/Engine/Content: 'F:/4.20.3-customEngine/Engine/Content' is outside repository

**Comment by Obamium69** [comment]
Is there any update on this issue, since I'm experience it using Linux  running  UE 5.3.2?

**Comment by MacLove13** [comment]
Any news?

```fatal: E:/Epic Games/UE_5.5/Engine/Plugins/Experimental/Animation/SkeletalMeshModelingTools/Content: 'E:/Epic Games/UE_5.5/Engine/Plugins/Experimental/Animation/SkeletalMeshModelingTools/Content' is outside repository at 'D:/Epic/GameAnimationSample'
fatal: E:/Epic Games/UE_5.5/Engine/Plugins/Experimental/MetaHuman/MetaHumanSDK/Content: 'E:/Epic Games/UE_5.5/Engine/Plugins/Experimental/MetaHuman/MetaHumanSDK/Content' is outside repository at 'D:/Epic/GameAnimationSample'
fatal: E:/Epic Games/UE_5.5/Engine/Plugins/Developer/Concert/ConcertSync/ConcertSyncClient/Content: 'E:/Epic Games/UE_5.5/Engine/Plugins/Developer/Concert/ConcertSync/ConcertSyncClient/Content' is outside repository at 'D:/Epic/GameAnimationSample'
fatal: E:/Epic Games/UE_5.5/Engine/Content: 'E:/Epic Games/UE_5.5/Engine/Content' is outside repository at 'D:/Epic/GameAnimationSample'
fatal: E:/Epic Games/UE_5.5/Engine/Plugins/Enterprise/DatasmithContent/Content/Materials/AliasReference.uasset: 'E:/Epic Games/UE_5.5/Engine/Plugins/Enterprise/DatasmithContent/Content/Materials/AliasReference.uasset' is outside repository at 'D:/Epic/GameAnimationSample'
fatal: E:/Epic Games/UE_5.5/Engine/Plugins/Experimental/Water/Content/Materials/Debug/Single_Gerstner_Wave.uasset: 'E:/Epic Games/UE_5.5/Engine/Plugins/Experimental/Water/Content/Materials/Debug/Single_Gerstner_Wave.uasset' is outside repository at 'D:/Epic/GameAnimationSample'```

**Comment by Skjol-Dev** [comment]
its been 6 years and doesnt seem to be fixed.
